utilities/transactions/write_unprepared_txn.cc

```diff
@@ -106,7 +106,10 @@ Status WriteUnpreparedTxn::HandleWrite(std::function<Status()> do_write) {
       largest_validated_seq_ =
           std::max(largest_validated_seq_, snapshot_->GetSequenceNumber());
     } else {
-      largest_validated_seq_ = kMaxSequenceNumber;
+      // TODO(lth): We should use the same number as tracked_at_seq in TryLock,
+      // because what is actually being tracked is the sequence number at which
+      // this key was locked at.
+      largest_validated_seq_ = db_impl_->GetLastPublishedSequence();
```

Comment:
Shall we also update the variable inline comment in https://github.com/facebook/rocksdb/blob/04a849b7b408e05d55557a16d1d677b80c543b30/utilities/transactions/write_unprepared_txn.h#L244